### PR TITLE
chore(test): Add test pkgs and integ tests for core

### PR DESCRIPTION
**Pull request by dnys1**
*Issue #, if available:*

*Description of changes:*
- Added integ tests for Amplify core to test configuration loading scenarios (must be run with Xcode)
- Added `amplify_test` package for common test utilities

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_flutter/example/pubspec.yaml [hunk 1]
```diff
@@ -21,9 +21,18 @@ dependencies:
     sdk: flutter
 
 dev_dependencies:
+  amplify_analytics_pinpoint: ^0.2.4
```
Comment (ragingsquirrel3) [inline packages/amplify_flutter/example/pubspec.yaml:24]: Should these versions be changed to match RC version?

File: packages/amplify_flutter/example/integration_test/config_test.dart [hunk 2]
```diff
@@ -0,0 +1,282 @@
+import 'dart:io';
```
Comment (ragingsquirrel3) [inline packages/amplify_flutter/example/integration_test/config_test.dart:1]: Should this (and all new dart files in this PR) have the big header with copyright info?

File: packages/amplify_flutter/example/pubspec.yaml (hunk 1 above)
Comment (Jordan-Nelson) [reply packages/amplify_flutter/example/pubspec.yaml:24]: I think relative paths would be consistent with the other example projects. Not sure if there is a reason for this to be different?

Comment (dnys1) [reply packages/amplify_flutter/example/pubspec.yaml]: Fixed

File: packages/amplify_flutter/example/integration_test/config_test.dart (hunk 2 above)
Comment (dnys1) [reply packages/amplify_flutter/example/integration_test/config_test.dart:16]: Fixed

**Review by Jordan-Nelson — APPROVED** [review]

**Review by ragingsquirrel3 — APPROVED** [review]

**Pull request merged**